src/main/java/myapp/HelperBuddy/Deadline.java

```diff
@@ -0,0 +1,85 @@
+package myapp.helperbuddy;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Deadline class representing a Deadline task
+ */
+public class Deadline extends Task {
+    private final LocalDateTime deadlineBy;
+
+    /**
+     * Constructs a {@code Deadline} task with the specified description and deadline.
+     *
+     * @param description The description of the task.
+     * @param deadlineBy  The deadline date and time for the task.
+     */
+    public Deadline(String description, LocalDateTime deadlineBy) {
+        super(description);
+        this.deadlineBy = deadlineBy;
+    }
+
+    /**
+     * Parses a string to create a Deadline task.
+     * The string should be in the format used for saving to a file,
+     * with fields separated by " | ".
+     * The fields are: task type ("D"), completion status (1 for done, 0 for not done),
+     * description, and optionally, the deadline.
+     * If the deadline is missing or incorrectly formatted, a warning is printed, and the deadline will be null.
+     *
+     * @param taskData The string representing the Deadline task.
```

Comment:
I like how detailed the comment is. Given that the comment covered the details of the parameter quite comprehensively, maybe the @param line here could be omitted?

Same sentiment on some of the other method comments